src/parser/lval.js

```diff
@@ -378,4 +380,18 @@ export default class LValParser extends NodeUtils {
       }
     }
   }
+
+  checkToRestConversion(node: SpreadElement): void {
+    const validArgumentTypes = [
```

Comment:
this should also be specified imho by the [RestElement's type](https://github.com/babel/babylon/blob/88cf1bc140d19da19e9496b6293103d992bd9e85/src/types.js#L594) but when I was trying to change it flow has yelled at me so much, that i just fled with my tails between my legs (for the time being - would love to adjust it properly)